
syntax = "proto3";

message NullMessage {}

message Step {
    fixed64 n = 1;
}

message NdArray {
    string dtype = 1;
    repeated int32 shape = 2;
    bytes data = 3;
}

message ScalarMetric {
    string name = 1;
    double y = 2;
    message Arg {
        double value = 1;
    }
    Arg x = 3;
}

service ParameterServer {
    rpc IncrementGlobalT(NullMessage) returns (Step) {}
    rpc ApplyGradients(stream NdArray) returns (NullMessage) {}
    rpc GetValues(NullMessage) returns (stream NdArray) {}
    rpc StoreScalarMetric(ScalarMetric) returns (NullMessage) {}
}
